### DISCUSSION TITLE
The type RegCloseKeySafeHandle generated for RegLoadAppKey() is not compatible with RegistryKey.FromHandle()

### DISCUSSION THREAD

**Discussion by efraimbart** [microsoft/CsWin32#440] [Q&A] [answered]
The type `RegCloseKeySafeHandle` which is generated as the out for `RegLoadAppKey()` does not inherit from `SafeRegistryHandle` and is therefore not compatible with `RegistryKey.FromHandle()`.

I'm kinda new to the whole Win32 API, so any help to figure where I should be going from here would be greatly appreciated!

**Comment by AArnott** [comment, marked as answer]
I see that `SafeRegistryHandle` is implemented to call `RegCloseKey` to release the handle, so that meets the purpose. We should be able to reuse that .NET type rather than generating our own.

**Reply by AArnott** [reply]
Fixed in #442
